src/sql/workbench/contrib/assessment/common/asmtActions.ts

```diff
@@ -225,3 +238,60 @@ export class AsmtSamplesLinkAction extends Action {
 		return this._openerService.open(URI.parse(AsmtSamplesLinkAction.configHelpUri));
 	}
 }
+
+export class AsmtGenerateHTMLReportAction extends Action {
+	public static readonly ID = 'asmtaction.generatehtmlreport';
+	public static readonly LABEL = nls.localize('asmtaction.generatehtmlreport', "Make HTML Report");
+	public static readonly ICON = 'bookreport';
+
+	constructor(
+		@IFileService private _fileService: IFileService,
+		@IOpenerService private _openerService: IOpenerService,
+		@IEnvironmentService private _environmentService: IEnvironmentService,
+		@IDialogService private readonly _dialogService: IDialogService,
+		@IClipboardService private readonly _clipboardService: IClipboardService,
+		@IAdsTelemetryService private _telemetryService: IAdsTelemetryService,
+		@INotificationService private readonly _notificationService: INotificationService
+	) {
+		super(AsmtGenerateHTMLReportAction.ID, AsmtGenerateHTMLReportAction.LABEL, AsmtGenerateHTMLReportAction.ICON);
+	}
+
+	public async run(context: IAsmtActionInfo): Promise<boolean> {
+		const fileName = this.generateReportFileName(new Date(context.component.recentResult.dateUpdated));
+		const filePath = path.join(this._environmentService.userRoamingDataHome.fsPath, 'SqlAssessmentReports', fileName);
```

Comment:
You don't want to ask them where to save it?

Reply:
That's a really good point. I have added a SaveAs dialog to ask a user for a file location.